boards/arm/stm32/common/src/stm32_nunchuck.c

```diff
@@ -0,0 +1,111 @@
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+
+#include <stdint.h>
+#include <fcntl.h>
+#include <errno.h>
+#include <debug.h>
+
+#include <nuttx/irq.h>
+#include <nuttx/arch.h>
+#include <nuttx/fs/fs.h>
+#include <nuttx/input/nunchuck.h>
+
+#include "stm32_i2c.h"
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Private Types
+ ****************************************************************************/
+
+/****************************************************************************
+ * Private Function Prototypes
+ ****************************************************************************/
+
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+/****************************************************************************
+ * Public Data
+ ****************************************************************************/
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: board_nunchuk_initialize
+ *
+ * Description:
+ *   Initialize and register the Nunchuck joystick driver driver.
+ *
+ * Input Parameters:
+ *   devno - The device number, used to build the device path as
+ *           /dev/nunchuckN
+ *   busno - The I2C bus number
+ *
+ * Returned Value:
+ *   Zero (OK) on success; a negated errno value on failure.
+ *
+ ****************************************************************************/
+
+int board_nunchuck_initialize(int devno, int busno)
+{
+  FAR struct i2c_master_s *i2c;
+  char devpath[14];
+  int ret;
+
+  iinfo("Initializing Wii Nunchuck!\n");
+
+  /* Initialize I2C */
+
+  i2c = stm32_i2cbus_initialize(busno);
+  if (i2c == NULL)
+    {
+      return -ENODEV;
+    }
+
+  /* Register the joystick device as /dev/nunchuck0 */
+
+  iinfo("Initialize joystick driver: %s\n", devname);
+
+  snprintf(devpath, 14, "/dev/nunchuck%d", devno);
+  ret = nunchuck_register(devname, i2c);
```

Comment:
```suggestion
    snprintf(devpath, 15, "/dev/nunchuck%d", devno);
    iinfo("Initialize joystick driver: %s\n", devpath);

  ret = nunchuck_register(devpath, i2c);
```
`devname` is not defined in this scope, it was used as a parameter in the old code.
Also 15 is the minimum to have "/dev/nunchuckX"

Reply:
Good catch. The string array also needs to be bigger then. Will change that.